pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -276,12 +278,8 @@ func (f *framework) RunScorePlugins(pc *PluginContext, pod *v1.Pod, nodes []*v1.
 	errCh := schedutil.NewErrorChannel()
 	workqueue.ParallelizeUntil(ctx, 16, len(nodes), func(index int) {
 		for _, pl := range f.scorePlugins {
-			weight, weightExists := f.pluginNameToWeightMap[pl.Name()]
-			if !weightExists {
-				err := fmt.Errorf("weight does not exist for plugin %v", pl.Name())
-				errCh.SendErrorWithCancel(err, cancel)
-				return
-			}
+			// Score plugins' weight has been checked when they are initialized.
+			weight, _ := f.pluginNameToWeightMap[pl.Name()]
```

Comment:
Practically this is ok, in theory it is not because Name() could return something different here than when we initialized the map, but I think this is a risk we can take :)

Reply:
Yes, this is a little problem. We could add a check that the registered name and the plugin name must be same if needed.

Reply:
It has been checked at L123, so it will not be a problem.